cdc/puller/puller.go

```diff
@@ -240,11 +241,15 @@ func (p *pullerImpl) Run(ctx context.Context) error {
 					continue
 				}
 				lastResolvedTs = resolvedTs
-				err := output(&model.RawKVEntry{CRTs: resolvedTs, OpType: model.OpTypeResolved, RegionID: e.RegionID})
+				ok, err := output(&model.RawKVEntry{CRTs: resolvedTs, OpType: model.OpTypeResolved, RegionID: e.RegionID})
 				if err != nil {
 					return errors.Trace(err)
 				}
-				atomic.StoreUint64(&p.resolvedTs, resolvedTs)
+				// If the resolved ts is not sent to output channel, it means the resolved ts is
+				// not advanced, so we should not update the p.resolvedTs.
+				if ok {
+					atomic.StoreUint64(&p.resolvedTs, resolvedTs)
+				}
```

Comment:
Does the same problem exist in master branch?

Reply:
yes, I will fix it in another PR.